aws.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/xuri/excelize/v2"
+)
+
+const (
+	region string = "us-east-1"
+)
+
+func createS3Client(region string) (*s3.Client, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region))
+	if err != nil {
+		return nil, err
+	}
+
+	return s3.NewFromConfig(cfg), nil
+}
+
+type S3ClientAPI interface {
```

Comment:
neat, this led me to learn about the recommended way of mocking in v2 of the API and why it has changed from v1